src/pip/_internal/commands/uninstall.py

```diff
@@ -59,6 +64,20 @@ def run(self, options, args):
 
         reqs_to_uninstall = {}
         for name in args:
+            if is_archive_file(name):
+                if os.path.isfile(name):
```

Comment:
I don't think you should try to reinvent `install_req_from_line()`.

It is called right below on line 81.

Currently, we do:
```
            if req.name:
                reqs_to_uninstall[canonicalize_name(req.name)] = req
```

I'd say that all that is needed would be to add an `else:` clause there and either log a warning or raise an error.

Reply:
@xavfernandez, I think I got what you're saying but I'm not sure. 
The logic of checking the validity of the package to uninstall is already implemented in `install_req_from_line()` and further manipulation to the `name` variable is not needed. Is that right?
Anyway, I tried to change the code according to my understanding of your comment. 😃

Reply:
@xavfernandez, I believe I got what you're saying but I'm not sure.
I refactored the code according to my understanding of how we should utilize the functionality of `install_req_from_line()`, it seems that it broadens the cases in which the error is raised at the cost of a specific error message. 
lmk what you think! 😄

Reply:
You understood me correctly :+1: :smile: